### PR TITLE
in Posets.m2, implement the Coxeter Polynomial

### PR DESCRIPTION
**Pull request by fchapoton**
which is one useful invariant

### CODE REVIEW

File: M2/Macaulay2/packages/Posets.m2 [hunk 1]
```diff
@@ -1571,6 +1572,15 @@ zetaPolynomial Poset := RingElement => opts -> P -> (
     sum(#X, i -> Y_i * product(drop(X, {i,i}), xj -> (R_0 - xj)/(X_i-xj)))
     )
 
+coxeterPolynomial = method(Options => {symbol VariableName => getSymbol "t"})
+coxeterPolynomial Poset := RingElement => opts -> P -> (
+    R := ZZ(monoid [opts.VariableName]);
+    M := P.RelationMatrix;
+    n := numrows M;
+    C = -M * inverse transpose M;
```
Comment (mahrud) [inline M2/Macaulay2/packages/Posets.m2:1580]: You need `:=` here.

Comment (fchapoton) [reply M2/Macaulay2/packages/Posets.m2]: Thanks. I have fixed that. Let us see.

**Comment by fchapoton** [comment]
Maybe this one should also wait for the opinion of authors of the package ?

**Comment by DanGrayson** [comment]
Dear authors ( @dcookii , Mapes , Whieldon ) : 

Are these changes okay with you?

**Comment by dcookii** [comment]
Fine with me.  Thanks @fchapoton for expanding Posets.

**Comment by fchapoton** [comment]
Is this considered validated by the authors, or do you expect every author to approve ?

**Comment by gwynwhieldon** [comment]
I'd also support this. I'm not sure if Sonja has seen it yet or not, but it
seems like a good addition to the current Posers package to me as well.

Best,

Gwyn

**Comment by DanGrayson** [comment]
Well, that's 2/3 of the authors, so I think we can go ahead.  (I sent Sonja an email message 4 days ago about it.)

**Pull request merged**